tests/test_change.cpp

```diff
@@ -0,0 +1,266 @@
+#include <gtest/gtest.h>
+#include <plugin_api.h>
+#include <config_category.h>
+#include <filter_plugin.h>
+#include <filter.h>
+#include <string.h>
+#include <string>
+#include <rapidjson/document.h>
+#include <reading.h>
+#include <reading_set.h>
+#include <logger.h>
+
+using namespace std;
+using namespace rapidjson;
+
+extern "C"
+{
+	PLUGIN_INFORMATION *plugin_info();
+	void plugin_ingest(void *handle,
+					   READINGSET *readingSet);
+	PLUGIN_HANDLE plugin_init(ConfigCategory *config,
+							  OUTPUT_HANDLE *outHandle,
+							  OUTPUT_STREAM output);
+	int called = 0;
+
+	void Handler(void *handle, READINGSET *readings)
+	{
+		called++;
+		*(READINGSET **)handle = readings;
+	}
+};
+
+
+TEST(CHANGE, configuration)
+{
+	// Test case : Configuration is wokring correctly
+
+	PLUGIN_INFORMATION *info = plugin_info();
+	ConfigCategory *config = new ConfigCategory("change", info->config);
+	ASSERT_NE(config, (ConfigCategory *)NULL);
+	config->setItemsValueFromDefault();
+	ASSERT_EQ(config->itemExists("enable"), true);
+	config->setValue("asset", "test");
+	config->setValue("trigger", "test");
+	config->setValue("enable", "true");
+
+	ReadingSet *outReadings;
+	void *handle = plugin_init(config, &outReadings, Handler);
+	vector<Reading *> *readings = new vector<Reading *>;
+	
+	long testValue1 = 10;
+	DatapointValue dpv1(testValue1);
+	Datapoint *value1 = new Datapoint("test", dpv1);
+	Reading *in1 = new Reading("test", value1);
+	readings->push_back(in1);
+
+	long testValue2 = 11;
+	DatapointValue dpv2(testValue2);
+	Datapoint *value2 = new Datapoint("test", dpv2);
+	Reading *in2 = new Reading("test", value2);
+	readings->push_back(in2);
+
+	long testValue3 = 12;
+	DatapointValue dpv3(testValue3);
+	Datapoint *value3 = new Datapoint("test", dpv3);
+	Reading *in3 = new Reading("test", value3);
+	readings->push_back(in3);
+
+	long testValue4 = 13;
+	DatapointValue dpv4(testValue4);
+	Datapoint *value4 = new Datapoint("test", dpv4);
+	Reading *in4 = new Reading("test", value4);
+	readings->push_back(in4);
+
+	long testValue5 = 14;
+	DatapointValue dpv5(testValue5);
+	Datapoint *value5 = new Datapoint("test", dpv5);
+	Reading *in5 = new Reading("test", value5);
+	readings->push_back(in5);
+
+	ReadingSet *readingSet = new ReadingSet(readings);
```

Comment:
Memory allocations on heap don't need to be freed?

Reply:
fixed